devtools/maven/src/main/java/io/quarkus/maven/DevMojo.java

```diff
@@ -242,18 +243,24 @@ public void execute() throws MojoFailureException, MojoExecutionException {
                     MavenProject mavenProject = projectMap.get(String.format("%s:%s:%s",
                             appArtifact.getGroupId(), appArtifact.getArtifactId(), appArtifact.getVersion()));
                     // no information about this project from Maven. Skip.
-                    if (mavenProject == null) {
-                        continue;
-                    }
-                    String sourcePath = null;
+
+                    String projectDirectory = null;
+                    List<String> sourcePaths = null;
                     String classesPath = null;
                     String resourcePath = null;
 
-                    List<String> sourcePaths = mavenProject.getCompileSourceRoots().stream()
-                            .map(Paths::get)
-                            .filter(Files::isDirectory)
-                            .map(src -> src.toAbsolutePath().toString())
-                            .collect(Collectors.toList());
+                    if (mavenProject == null) {
+                        projectDirectory = localProject.getRawModel().getProjectDirectory().getPath();
```

Comment:
Let's use `localProject.getDir()` instead. `localProject.getRawModel().getProjectDirectory().getPath()` relies on the presence of `pomFile` in the model, which most probably will be there but if for some reason it's not, it's going to return null. `localProject.getDir()` is going to work for sure.

Reply:
done